commit e210101bbc28be6449eb418c4650b668b42381db
Author: Julien <Julien@.(none)>
Date:   Thu Jul 9 21:27:40 2009 -0400

    Patch for static compilation into stdcall dll

diff --git a/src/cairo-font-face-twin.c b/src/cairo-font-face-twin.c
index 3f82733..934a356 100644
--- a/src/cairo-font-face-twin.c
+++ b/src/cairo-font-face-twin.c
@@ -299,7 +299,7 @@ twin_font_face_create_properties (cairo_font_face_t *twin_face,
 
     status = cairo_font_face_set_user_data (twin_face,
 					    &twin_properties_key,
-					    props, free);
+					    props, free_data);
     if (unlikely (status)) {
 	free (props);
 	return status;
@@ -444,7 +444,7 @@ twin_scaled_font_compute_properties (cairo_scaled_font_t *scaled_font,
     /* Save it */
     status = cairo_scaled_font_set_user_data (scaled_font,
 					      &twin_properties_key,
-					      props, free);
+					      props, free_data);
     if (unlikely (status))
 	goto FREE_PROPS;
 
diff --git a/src/cairo-hull.c b/src/cairo-hull.c
index 1fa919b..d35d893 100644
--- a/src/cairo-hull.c
+++ b/src/cairo-hull.c
@@ -85,7 +85,7 @@ _slope_length (cairo_slope_t *slope)
 			     _cairo_int32x32_64_mul (slope->dy, slope->dy));
 }
 
-static int
+static int __cdecl
 _cairo_hull_vertex_compare (const void *av, const void *bv)
 {
     cairo_hull_t *a = (cairo_hull_t *) av;
diff --git a/src/cairo-png.c b/src/cairo-png.c
index d4f0476..03fafee 100644
--- a/src/cairo-png.c
+++ b/src/cairo-png.c
@@ -97,7 +97,7 @@ convert_data_to_bytes (png_structp png, png_row_infop row_info, png_bytep data)
  * stderr and rely on the user to check for errors via the #cairo_status_t
  * return.
  */
-static void
+static void __stdcall
 png_simple_error_callback (png_structp png,
 	                   png_const_charp error_msg)
 {
@@ -114,7 +114,7 @@ png_simple_error_callback (png_structp png,
     /* if we get here, then we have to choice but to abort ... */
 }
 
-static void
+static void __stdcall
 png_simple_warning_callback (png_structp png,
 	                     png_const_charp error_msg)
 {
@@ -670,7 +670,7 @@ read_png (struct png_read_closure_t *png_closure)
 					  CAIRO_MIME_TYPE_PNG,
 					  mime_data,
 					  mime_data_length,
-					  free,
+					  free_data,
 					  mime_data);
     if (unlikely (status)) {
 	free (mime_data);
