commit e95254031111db96653711c058e4ba9efc5f192c
Author: unknown <Julien@.(none)>
Date:   Thu Jun 4 11:04:43 2009 -0400

    Patch for libdir definition

diff --git a/pango/pango-utils.c b/pango/pango-utils.c
index 9801a30..c5c4dcc 100644
--- a/pango/pango-utils.c
+++ b/pango/pango-utils.c
@@ -721,7 +721,7 @@ pango_get_sysconf_subdirectory (void)
 G_CONST_RETURN char *
 pango_get_lib_subdirectory (void)
 {
-#ifdef G_OS_WIN32
+#if defined(G_OS_WIN32) && defined(LIBDIR)
   static gchar *result = NULL;
 
   if (result == NULL)
@@ -738,7 +738,7 @@ pango_get_lib_subdirectory (void)
     }
   return result;
 #else
-  return LIBDIR "/pango";
+  return "/pango";
 #endif
 }
 
